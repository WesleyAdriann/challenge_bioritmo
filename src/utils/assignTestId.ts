type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
)  => ({
    'data-testid': `${componentType}_${componentId}`,
  })


export const mockTestID = (
    componentType: componentType,
    componentId: string
  ) => assignTestId(componentType, componentId)
