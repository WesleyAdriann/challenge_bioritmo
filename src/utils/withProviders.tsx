
import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../style'

export const StorybookProviders: React.FC = ({
  children,
}) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
)
