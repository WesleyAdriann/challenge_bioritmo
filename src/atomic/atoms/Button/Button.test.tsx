import { render, fireEvent, cleanup} from '@testing-library/react'

import { mockTestID, StorybookProviders as wrapper } from '../../../utils'

import { Button, IButton } from './index'

describe('Atoms/Button', () => {
  const props: IButton = {
    testID: 'Button',
    onClick: jest.fn(),
    text: 'Text Test'
  }


  const renderComponent = () => render(
    <Button {...props} />, { wrapper }
  )

  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  const el_container = mockTestID('button', props.testID!)

  it(`should render component #${el_container}`, async () => {
    const sut = await component.findByTestId(el_container)
    expect(sut).toBeTruthy()
  })

  it('should render text prop', async () => {
    const sut = component.getByText(props.text)
    expect(sut).toBeTruthy()
  })

  it('should call onClick prop on press in button', async () => {
    const sut = await component.findByTestId(el_container)
    fireEvent.click(sut)
    expect(props.onClick).toBeCalled()
  })

  it('should render snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot()
  })

})
