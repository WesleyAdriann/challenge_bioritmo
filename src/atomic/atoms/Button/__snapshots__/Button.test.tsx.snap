// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Button should render snapshot 1`] = `
.c0 {
  background-color: #FFB612;
  border-radius: 4px;
  border-color: #FFB612;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  font-weight: bold;
  padding: 12px;
  text-align: center;
  text-transform: normal;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  word-wrap: break-word;
}

.c0:active {
  -webkit-filter: brightness(95%);
  filter: brightness(95%);
}

<div
  class="c0"
  data-testid="button_Button"
>
  Text Test
</div>
`;
